lib/nrf_modem_lib/Kconfig

```diff
@@ -56,6 +56,82 @@ config NRF_MODEM_LIB_SENDMSG_BUF_SIZE
 	  the repacked message would not fit into the buffer, `sendmsg` sends
 	  each message part separately.
 
+comment "Heap and buffers"
+
+config NRF_MODEM_LIB_HEAP_SIZE
+	int "Library heap size"
+	default 512
+	range 512 2048
+	help
+	  Size of the heap buffer used by the library.
+	  This heap is allocated in the application's RAM.
+
+config NRF_MODEM_LIB_SHMEM_CTRL_SIZE
+	hex
+	default NRF_MODEM_SHMEM_CTRL_SIZE
+	help
+	  Size of the shared memory area used for control structures.
+	  This is a constant for a given library build, and is exported
+	  by the library via NRF_MODEM_SHMEM_CTRL_SIZE.
+
+config NRF_MODEM_LIB_SHMEM_TX_SIZE
+	int "TX region size"
+	range 1024 16384
+	default 4096
```

Comment:
We should discuss the default values here, currently this file https://github.com/nrfconnect/sdk-nrfxlib/blob/master/nrf_modem/include/nrf_modem_limits.h states that the max message size is 4096, but that is not possible with the current defaults. So I guess we should either have a bigger default, or we should make it clear in the documentation that the default value will not let you send messages this large.

Reply:
To make it not fail on such a message, the size must be 4164 bytes. I calculated it to 4160, but to make the allocator actually allocate 520 (520 * 8 = 4160 bytes) units you need that value.

Reply:
I am starting to think that this default should be increased a bit. Currently we are supporting 8 sockets (AT, UDP, TCP) + GNSS and DFU, and the current limit is actually a bit to small for the biggest packages from just one socket. It is important that all samples and typical use-cases work out of the box with the default settings. I think 8k for both TX and RX will be good number for the default value, and then I think it will a documentation job to explain the users  how to set these values and how to tune them using the diagnostics.

Reply:
I am fine with bumping up the size, but instead of doubling it would you be okay with +2k (50% more), i.e. 4k+2k = `6144` bytes? This way we can keep the default config comfortably below 16 kB total.

Reply:
I think it is better to just up this number 8k per direction so we have some margin to cover any user-case and then instead educate users through documentation how to optimize these sizes per use-cases. Remember that the size before was 28672 bytes. If we then take the  8 * 64 (RPC messages) + 8 * 8 (event slots) + app descriptor + modem descriptor = 640 bytes and remove that from 28672 - 640 =  27456 bytes. Compare the new memory footprint for TX and RX memory, 16384 bytes, against the old memory footprint that is 40% less. We also don't waste trace memory if the feature is not used.

So I would say that this is in any case a significant improvement. I think it is better to play it a bit safe and make sure we have enough margin for any use-case that the customers may throw at it.

Reply:
Ok, upped to 8k

Reply:
Thanks